src/components/areas/TopBarArea.js

```diff
@@ -129,5 +129,6 @@ class TopBarArea extends React.PureComponent {
   }
 }
 export default connect(state => ({
-  apiKey: state.getIn(['save', 'data', 'apiKey'])
+  apiKey: state.getIn(['save', 'data', 'apiKey']),
+  emuchat: state.getIn(['save', 'data', 'emuchat'])
```

Comment:
You don't need emuchat in this component. This line can be removed.

I know this is all new and a lot to learn.
Here is the doc:
https://redux.js.org/docs/basics/UsageWithReact.html

What you are basically doing here is injecting the component with a value from global state. Redux holds the global state and React is relevant for our virtual DOM. Since you don't need emuchat in the component, we don't have to inject the value from global state into this component.